### PR TITLE
Adds BankQuery::Supply support

### PR DESCRIPTION
**Pull request by JakeHartnell**
Builds off: https://github.com/CosmWasm/cw-multi-test/pull/17

### CODE REVIEW

**Comment by JakeHartnell** [comment]
@chipshort please have a look! 🙏

File: src/bank.rs [hunk 1]
```diff
@@ -61,12 +61,27 @@ impl BankKeeper {
...
-    // this is an "admin" function to let us adjust bank accounts
     fn get_balance(&self, bank_storage: &dyn Storage, account: &Addr) -> AnyResult<Vec<Coin>> {
         let val = BALANCES.may_load(bank_storage, account)?;
         Ok(val.unwrap_or_default().into_vec())
     }
 
+    fn get_supply(&self, bank_storage: &dyn Storage, denom: String) -> AnyResult<Coin> {
+        let supply: Uint128 = BALANCES
+            .range(bank_storage, None, None, Order::Ascending)
+            .map(|a| a.unwrap().1)
```
Comment (chipshort) [inline src/bank.rs:72]: I'd still like to get rid of this `unwrap`. There's multiple ways to do it, but I'd just collect into a `StdResult<Vec<_>>` and continue with that:
`BALANCES.range(...).collect::<StdResult<Vec<_>>>()?.map(...).fold(...)`.

File: src/bank.rs [hunk 2]
```diff
@@ -296,6 +317,13 @@ mod test {
         let raw = bank.query(&api, &store, &querier, &block, req).unwrap();
         let res: BalanceResponse = from_slice(&raw).unwrap();
         assert_eq!(res.amount, coin(0, "eth"));
+
+        let req = BankQuery::Supply {
+            denom: "eth".into(),
+        };
+        let raw = bank.query(&api, &store, &querier, &block, req).unwrap();
+        let res: SupplyResponse = from_slice(&raw).unwrap();
+        assert_eq!(res.amount, coin(100, "eth"));
```
Comment (chipshort) [inline src/bank.rs:326]: Nice test. Another case where the amount is owned by multiple accounts would be nice.

**Review by chipshort — COMMENTED** [review]
Thanks for improving this. Let's avoid that unwrap, then we can get this in.

File: src/bank.rs (hunk 1 above)
Comment (JakeHartnell) [reply src/bank.rs:72]: Cool, will do.

File: src/bank.rs (hunk 2 above)
Comment (JakeHartnell) [reply src/bank.rs:326]: Great! Will add.

Comment (JakeHartnell) [reply src/bank.rs:330]: Done!

**Comment by JakeHartnell** [comment]
@chipshort, have another look! No unwrap, and added a test case for total supply with multiple accounts.

**Review by chipshort — APPROVED** [review]
LGTM

**Pull request merged**